blockchain/indexers/utreexobackend.go

```diff
@@ -157,111 +135,180 @@ func (idx *FlatUtreexoProofIndex) FlushUtreexoState() error {
 	if err != nil {
 		return err
 	}
-	err = idx.utreexoState.state.WriteForestToDisk(forestFile, true, false)
-	if err != nil {
-		return err
-	}
-
-	miscFilePath := filepath.Join(basePath, defaultUtreexoMiscFileName)
-	// write other misc forest data
-	miscForestFile, err := os.OpenFile(
-		miscFilePath, os.O_RDWR|os.O_CREATE, 0666)
-	if err != nil {
-		return err
-	}
-	err = idx.utreexoState.state.WriteMiscData(miscForestFile)
+	_, err = idx.utreexoState.state.WriteTo(forestFile)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-// restoreUtreexoState restores forest fields based off the existing utreexo state
-// on disk.
-func restoreUtreexoState(cfg *UtreexoConfig, basePath string) (
-	*accumulator.Forest, error) {
+func serializeUtreexoRoots(p *utreexo.Pollard) ([]byte, error) {
```

Comment:
Should get rid of this as it's a duplicate of `SerializeUtreexoRoots` in `blockchain`